keras_contrib/applications/densenet.py

```diff
@@ -230,32 +241,31 @@ def DenseNetFCN(input_shape, nb_dense_block=5, growth_rate=16, nb_layers_per_blo
 
     upsampling_type = upsampling_type.lower()
 
-    if upsampling_type not in ['upsampling', 'deconv', 'atrous', 'subpixel']:
-        raise ValueError('Parameter "upsampling_type" must be one of "upsampling", '
-                         '"deconv", "atrous" or "subpixel".')
+    if upsampling_type not in ['upsampling', 'deconv', 'subpixel']:
+        raise ValueError('Parameter "upsampling_type" must be one of '
+                         '"upsampling", "deconv", or "subpixel".')
 
     if upsampling_type == 'deconv' and batchsize is None:
         raise ValueError('If "upsampling_type" is deconvoloution, then a fixed '
                          'batch size must be provided in batchsize parameter.')
 
     if input_shape is None:
-        raise ValueError('For fully convolutional models, input shape must be supplied.')
+        raise ValueError(
+            'For fully convolutional models, input shape must be supplied.')
 
     if type(nb_layers_per_block) is not list and nb_dense_block < 1:
         raise ValueError('Number of dense layers per block must be greater than 1. Argument '
                          'value was %d.' % (nb_layers_per_block))
 
-    if upsampling_type == 'atrous':
-        warnings.warn('Atrous Convolution upsampling does not correctly work (see https://github.com/fchollet/keras/issues/4018).\n'
-                      'Switching to `upsampling` type upscaling.')
-        upsampling_type = 'upsampling'
-
     # Determine proper input shape
     input_shape = _obtain_input_shape(input_shape,
                                       default_size=32,
                                       min_size=16,
-                                      data_format=K.image_dim_ordering(),
-                                      include_top=include_top)
+                                      data_format=K.image_data_format(),
+                                      # If doing segmentation we still include top
+                                      # but _obtain_input_shape only supports
+                                      # labeling, not segmentation networks.
+                                      include_top=False)
```

Comment:
You could wait for my bugfix PR to be merged, it removes this dependency on _obtain_input_shape and uses only those parts of it which make sense to segmentation models. If not, atleast remove the comments inlined with the method arguments and put it above the method call.